### DISCUSSION TITLE
URI-scheme for Shortcuts

### DISCUSSION THREAD

**Discussion by bhtg** [criticalmaps/criticalmaps-ios#493] [Ideas]
Thank you all for the great work! The app is very slick and I use it multiple times a week.

I was wondering if there are any plans to deactivate the location sharing automaticly?

- My favourite implementation would be to link the observer mode switch to a URI-scheme so we can create CM shortcuts.
- A timer might be a simpler version but this could also be achieved with the URI-scheme combined with a Siri-Shortcut.
- A geofence would be another way to create more privacy, but I guess it is difficult to implement.

Could Apple's app review team have anything against this feature?

Unfortunately I don't have the experience to do this myself, but I'm happy to help test if that helps.

**Comment by genevievecory** [comment]
? I don't understand the request.

Do you want CriticalMaps to start in Observer mode?

**Reply by bhtg** [reply]
Thanks for the fast reply. 
I regularly forget to turn the observer mode back on after a CM or demo, and sometimes I see that the app has been running in the background all night. I would love to have an option to help users not leak their address after an event. I thought the ability to toggle the observer mode via Siri shortcuts would help with that. I hope this makes it a bit clearer.

**Comment by mltbnz** [comment]
Hey. I think the idea is good and also the way to solve it via a deeplink.

I'll might give it a shot when there is some time.

**Reply by bhtg** [reply]
Great! This could even indirectly solve the "start in observermode" idea from @genevievecory .
